sdcm/cluster.py

```diff
@@ -641,6 +642,44 @@ def _get_coredump_backtraces(self, last=True):
             self.log.error('Error retrieving core dump backtraces : %s',
                            details)
 
+    def _upload_coredump(self, coredump):
+        base_upload_url = 'scylladb-users-upload.s3.amazonaws.com/%s/%s'
+        coredump_id = os.path.basename(coredump)[:-3]
+        upload_url = base_upload_url % (coredump_id, os.path.basename(coredump))
+        self.log.info('Uploading coredump %s to %s' % (coredump, upload_url))
+        self.remoter.run("sudo curl --request PUT --upload-file "
+                         "'%s' '%s'" % (coredump, upload_url))
+        self.log.info("To download it, you may use "
+                      "'curl --user-agent [user-agent] %s > %s'",
+                      upload_url, os.path.basename(coredump))
+
+    def _get_coredump_size(self, coredump):
+        try:
+            res = self.remoter.run('stat -c %s {}'.format(coredump))
+            return int(res.stdout.strip())
+        except Exception as ex:
+            self.log.error('Failed getting coredump file size: %s', ex)
+        return None
+
+    def _split_coredump(self, coredump):
+        core_files = []
+        try:
+            self.remoter.run('sudo yum install -y pigz')
+            self.remoter.run('sudo pigz --fast {}'.format(coredump))
+            file_name = '{}.gz'.format(coredump)
+            core_files.append(file_name)
+            file_size = self._get_coredump_size(file_name)
+            if file_size and file_size > COREDUMP_MAX_SIZE:
+                cnt = file_size / COREDUMP_MAX_SIZE
+                cnt += 1 if file_size % COREDUMP_MAX_SIZE > 0 else cnt
+                self.log.debug('Splitting coredump to {} files'.format(cnt))
+                res = self.remoter.run('sudo split -n {} {} {}.;ls {}.a*'.format(
```

Comment:
It will bring the compressed source file too(coredump_name.gz) which is unwanted.
I think, we'll not reach this count, because the compressed file size should be not so big.